bin/utest/sigaction.c

```diff
@@ -45,3 +53,47 @@ int test_sigaction_handler_returns(void) {
 
   return 0;
 }
+
+#define NCHLD 7
+
+static volatile int nreaped;
+
+static void sigchld_handler(int signo, siginfo_t *si, void *uctx) {
+  assert(si->si_code == CLD_EXITED);
+
+  printf("child=%d (uid=%d) sent signal info\n", si->si_pid, si->si_uid);
+
+  while (waitpid(-1, NULL, WNOHANG) > 0)
+    nreaped++;
+}
+
+int test_sigaction_siginfo_from_children(void) {
+  struct sigaction sa;
+
+  memset(&sa, 0, sizeof(sa));
+  sa.sa_sigaction = sigchld_handler;
+  sa.sa_flags = SA_SIGINFO;
+  assert(sigaction(SIGCHLD, &sa, NULL) == 0);
+
+  sigset_t set, old;
+  __sigemptyset(&set);
+  __sigaddset(&set, SIGCHLD);
+  assert(sigprocmask(SIG_BLOCK, &set, &old) == 0);
+
+  for (int i = 0; i < NCHLD; i++) {
+    pid_t pid = fork();
+    assert(pid >= 0);
+    if (!pid) {
+      sleep(i & 0x1);
```

Comment:
Don't use sleep in tests, unless there's really no other choice. I don't think you absolutely need it here. We prefer `sched_yield` in similar situations.